scanner_test.go

```diff
@@ -450,6 +453,67 @@ func TestDecode(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("wrap errors correctly", func(t *testing.T) {
+
+		t.Run("wrap error from Decoder", func(t *testing.T) {
+			// Use a int parse error as the type example
+			decoder := ss.FuncTagDecoder(func(field ss.Field) (interface{}, error) {
+				return strconv.ParseInt("not-an-int", 10, 0)
+			})
+			err := ss.Decode(&struct {
+				A int `a:""`
+			}{}, decoder)
+
+			var parseErr *strconv.NumError
+			require.True(t, errors.As(err, &parseErr), "error %#v should wrap %T", err, parseErr)
+			require.Equal(t, parseErr.Err, strconv.ErrSyntax)
+		})
+
+		t.Run("wrap error from nested Decoder", func(t *testing.T) {
+			// Use a int parse error as the type example
+			var decoder ss.TagDecoder
+			decoder = ss.FuncTagDecoder(func(field ss.Field) (interface{}, error) {
+				if field.Name == "A" {
+					// Recurse
+					return decoder, nil
+				}
+				return strconv.ParseInt("not-an-int", 10, 0)
+			})
+
+			type Outer struct {
+				A struct{ B int }
+			}
+			err := ss.Decode(&Outer{}, decoder)
+
+			require.Error(t, err)
+			// Sanity check: the outer error _does_ contain the string we don't want to see in the wrapped error
+			require.Contains(t, err.Error(), "error decoding nested field")
```

Comment:
I am using a thin wrapper for testify on the other tests ([this one](https://github.com/VinGarcia/structscanner/tree/master/internal/testtools)).

There are two reasons for that:

One is that I think testify has too many functions, and they tempt developers to try to use it all, which makes the tests a little bit harder to read, since the dev reading it will often need to learn a new assert function each time. Often we'll just trust the name of the function which has already caused me to accept a PR that seemed to have a good test, but the test was actually not asserting anything. So it's an extra risk caused by an unnecessarily big API surface.

The other reason is that having a wrapper between me and an external dependency I can optimize some common use-cases like the one above, we have a function for that:

```suggestion
			// Sanity check: the outer error _does_ contain the string we don't want to see in the wrapped error
			tt.AssertErrContains(t, err, "error decoding nested field")
```

It can also be used to make a loosely couple assertion in regards to whether the error contains some key words that would be important for understanding what went wrong, e.g.:

```suggestion
			// Sanity check: the outer error _does_ contain the string we don't want to see in the wrapped error
			tt.AssertErrContains(t, err, "error decoding nested field", "A", "not-an-int")
```

Can you update this PR to use this wrapper instead of using the `require` package directly?

Reply:
Sure thing